compose/ui/ui-graphics/src/skikoMain/kotlin/androidx/compose/ui/graphics/SkiaBackedCanvas.skiko.kt

```diff
@@ -117,7 +117,15 @@ internal class SkiaBackedCanvas(val skia: org.jetbrains.skia.Canvas) : Canvas {
     }
 
     override fun drawRect(left: Float, top: Float, right: Float, bottom: Float, paint: Paint) {
-        skia.drawRect(SkRect.makeLTRB(left, top, right, bottom), paint.skia)
+        /*
+         * Do not use makeLTRB here because it adds additional checks (like left <= right) that
+         * might lead to a crash. It's not required here because skia sorts it internally.
+         * See https://github.com/JetBrains/skia/blob/skiko-m105/src/core/SkCanvas.cpp#L1776-L1778
+         *
+         * Please note that "negative" size is used in material components to support RTL.
```

Comment:
It is better to do the same in the other drawing methods (drawRoundRect, ...) to be consistent.

Also, it is better to remove the restrictions from Skiko itself, as Skia handles such values well by the contract. But we can do that later.

If there are doubts, please create a task.

Reply:
Disagree. We cannot safely drop this check from all places. I'd keep it only for ones where we're sure that it's handled by skia itself.

For example `drawRoundRect` that you mentioned uses skia's `drawRRect` that does [NOT](https://github.com/JetBrains/skia/blob/skiko-m105/src/core/SkCanvas.cpp#L1806-L1808) have this check.